Inverted printf specifiers result in improper output

<!--- STOP! Before you open an issue please search this repository's issues to see if it has already been reported. This helps reduce duplicate issues from being created. -->
These printf specifiers are inverted; this results in improper output:

https://github.com/adobe/zdw/blob/8013a42200b254c89a110bde393af2fd451206fe/cplusplus/includes.h#L28-L31
<!--- SECURITY DISCLOSURE: If this is a security disclosure please follow the guidelines in CONTRIBUTING.md. This helps keep folks from accidentally releasing vulnerabilities before the maintainers get a chance to fix the issue. -->

### Expected Behavior

Code should compile without warnings.  Printouts shouldn't contain appended "ll" or "l" depending on platform configuration.

### Actual Behavior

Program execution without quiet flag emits output like the following:

```
Reading 10485540ll byte dictionary
Reading 524160 rows
../test-files/movie_tickets.zdw.xz uncompressed
```

Compilation on platforms without `PRIu64` definition in `inttypes.h` and `unsigned long` is not 64-bit (such as a 32-bit linux with an older libc), produces warnings, such as the following, when `-Wall` is supplied:

```
UnconvertFromZDW.cpp:717:93: warning: format '%u' expects argument of type 'unsigned int', but argument 3 has type 'ULONGLONG {aka long long unsigned int}' [-Wformat=]
    fprintf(this->statusOutput, "Reading %" PF_LLU " byte dictionary\n", this->dictionarySize);
```

### Reproduction

a) Build the code on any platform.  Run code without the quiet flag, and observe the output contains extraneous output.
b) Build code on an older 32-bit linux with `-Wall`, observe warnings.

Neither should occur.

--- a/cplusplus/CMakeLists.txt
+++ b/cplusplus/CMakeLists.txt
@@ -1,7 +1,8 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.8)
 project(zdw)
 
 add_definitions(-D_FILE_OFFSET_BITS=64 -D__STDC_LIMIT_MACROS)
+add_compile_options("-Wall")
 
 add_library(zdw
 	UnconvertFromZDW.cpp
@@ -35,7 +36,7 @@ find_package(Threads REQUIRED)
 
 include_directories( zdw ${CMAKE_CURRENT_SOURCE_DIR} ${ZLIB_INCLUDE_DIRS} )
 
-# for cmake 2.6 compatibility, can't automatically handle include files
+# TODO: Investigate setting this flag on CMAKE 2.8+
 # target_include_directories( zdw PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} ${ZLIB_INCLUDE_DIRS} )
 
 target_link_libraries(zdw ${ZLIB_LIBRARIES})

--- a/cplusplus/includes.h
+++ b/cplusplus/includes.h
@@ -24,8 +24,7 @@
 #ifdef PRIu64
 #define PF_LLU PRIu64
 #else
-#include <limits.h>
-#if ULONG_MAX==1844674073709551615
+#if (_LP64 || __LP64__) // i.e., sizeof(uint64_t) == sizeof(unsigned long)
 #define PF_LLU "lu"
 #else
 #define PF_LLU "llu"

--- a/cplusplus/includes.h
+++ b/cplusplus/includes.h
@@ -26,9 +26,9 @@
 #else
 #include <limits.h>
 #if ULONG_MAX==1844674073709551615
-#define PF_LLU "ul"
+#define PF_LLU "lu"
 #else
-#define PF_LLU "ull"
+#define PF_LLU "llu"
 #endif
 #endif
 

